entrypoint.sh

```diff
@@ -0,0 +1,31 @@
+#!/bin/bash
+set -ex
+
+if [[ -z "$*" ]]; then
+  echo "Expecting one or more arguments"
+  exit 1
+fi
+
+#remove spack if exist previously
+if [[   -d "$GITHUB_WORKSPACE"/temp/spack ]]; then
+    rm -rf "$GITHUB_WORKSPACE"/temp
+fi
+
+# install Spack
+git clone https://github.com/spack/spack  "$GITHUB_WORKSPACE"/temp/spack&& \
+    PATH=$PATH:"$GITHUB_WORKSPACE"/temp/spack/bin && \
+    echo "$PATH" && export PATH && \
+    bash "$GITHUB_WORKSPACE"/temp/spack/share/spack/setup-env.sh
```

Comment:
this opens a new shell and the changes that `setup-env.sh` does to the environment do not persist, so we need to use `source` instead.